lib/MySQL_Session.cpp

```diff
@@ -1538,19 +1542,41 @@ bool MySQL_Session::handler_special_queries(PtrSize_t *pkt) {
 			return true;
 		}
 	}
-	if ( (pkt->size == 18) && (strncasecmp((char *)"SHOW WARNINGS",(char *)pkt->ptr+5,13)==0) ) {
-		SQLite3_result * resultset=new SQLite3_result(3);
-		resultset->add_column_definition(SQLITE_TEXT,"Level");
-		resultset->add_column_definition(SQLITE_TEXT,"Code");
-		resultset->add_column_definition(SQLITE_TEXT,"Message");
+	// if query digest is disabled, warnings in ProxySQL are also deactivated, 
+	// resulting in an empty response being sent to the client.
+	if ((pkt->size == 18) && (strncasecmp((char*)"SHOW WARNINGS", (char*)pkt->ptr + 5, 13) == 0) &&
```

Comment:
Since we check `CurrentQuery.QueryParserArgs.digest_text` , maybe it is the right opportunity to perform a string compare on the `digest_text` and not on the packet itself . This will allows us to match also in case of comments.
Also, maybe it makes sense to move all this code in `handler___status_WAITING_CLIENT_DATA___STATE_SLEEP___MYSQL_COM_QUERY_qpo()`

Reply:
This part will only be executed if digest_text is null ('mysql-query_digests'  is disabled)